tree/treeplayer/test/ttreereader_friends.cxx

```diff
@@ -214,3 +215,77 @@ TEST_F(TTreeReaderFriends, MainTreeShorterTChainExtraFriend)
                       /*matchFullMessage*/ true);
    r.SetEntry(2);
 }
+
+// ROOT-9886 "Reader: Unexpected value in friend tree: expected 13, read 3"
+TEST_F(TTreeReaderFriends, EntryChainFriend)
+{
+   constexpr const int kEntriesPerTree = 10;
+   for (int i = 0; i < 3; ++i) {
+      TFile file(std::string("maintree9886_" + std::to_string(i) + ".root").c_str(), "RECREATE");
+      TTree tree("tree", "Main tree");
+      int val;
+      tree.Branch("val", &val);
+      for (int j = 0; j < kEntriesPerTree; ++j) {
+         val = -(j + i * kEntriesPerTree);
+         tree.Fill();
+      }
+      tree.Write();
+   }
+   for (int i = 0; i < 1; ++i) {
+      TFile file(std::string("friendtree9886_" + std::to_string(i) + ".root").c_str(), "RECREATE");
+      TTree tree("friend", "Friend tree");
+      int entry;
+      tree.Branch("entry", &entry);
+      for (int j = 0; j < 3 * kEntriesPerTree; ++j) {
+         entry = j + i * kEntriesPerTree;
+         tree.Fill();
+      }
+      tree.Write();
+   }
+
+   // Test using traditional SetBranchAddress
+   {
+      std::unique_ptr<TChain> chainFriend{new TChain("friend")};
+      chainFriend->Add("friendtree9886_*.root");
+      std::unique_ptr<TChain> chainMain{new TChain("tree")};
+      chainMain->Add("maintree9886_*.root");
+      chainMain->AddFriend(chainFriend.get());
+      TChain *chain = chainMain.get();
+      EXPECT_EQ(chain->GetEntries(), 30);
+      EXPECT_EQ(chainFriend->GetEntries(), 30);
+
+      int mainVal;
+      int friendVal;
+      chain->SetBranchAddress("val", &mainVal);
+      chain->SetBranchAddress("entry", &friendVal);
+      int entry = -1;
+      while (chain->GetEntry(++entry)) {
+         EXPECT_EQ(mainVal, -entry);
+         EXPECT_EQ(friendVal, entry);
+      }
+   }
+
+   // Equivalent test using TTreeReader
+   {
+      // For some reason, if we reuse the chain from the main scope before
+      // it crashes when reaching entry number 10, dunno why, TODO debug
```

Comment:
Does this sentence mean there is still something to investigate regarding this issue?

Reply:
Thanks for the review! I would say this new coverage test discovered a separate issue (different). So I would vote for merging + closing the JIRA ticket and then reopening this newly discovered issue on GH.

Reply:
It's now https://github.com/root-project/root/issues/17843